### PR TITLE
Excluded files in python plugin #169, (Build)Requires as attribute #177, + fixes

### PR DESCRIPTION
**Pull request by cibo94**

### CODE REVIEW

**Comment by jsilhan** [comment]
You managed this task really well by testing and exploring just by yourself. Using queries instead Selectors was easier choice. I would erase `_strip_arch_n_os` and `_strip_version` and implement `_result` as:
```python
def _resolve(files, query):
    for _file in files:
        pkg = query.filter(file=_file)[0]
        yield pkg.name + " >= " + pkg.version
```

File: tests/test_plugins.py [hunk 1]
```diff
@@ -87,6 +96,18 @@ def test_python_find_requires(self):
...
 
+    def test_files_to_pkgs(self):
+        ftpp = FilesToPkgsPlugin()
+        self.spec.Requires = [
+            "/usr/lib/python3.4/site-packages/dnf/conf/read.py",
+            "/usr/lib/python3.4/site-packages/dnf/yum/sqlutils.py",
+            "/usr/lib/python3.4/site-packages/dnf/query.py"
+        ]
+        ftpp.installed("", self.spec, self.stack)
+        self.assertTrue(len(self.spec.Requires) > 0)
```
Comment (jsilhan) [inline tests/test_plugins.py]: You're doing `_uniq` so the require of one package should occur only once or not? I.e. do this:
`self.assertEqual(len(self.spec.Requires), 1); sefl.assertTrue(sel.spec.Requires[0].startswith("dnf >= "))`.

File: tests/test_plugins.py [hunk 2]
```diff
@@ -4,16 +4,25 @@
...
 from rpg.spec import Spec
+import dnf
+import re
 
 
 class FindPatchPluginTest(PluginTestCase):
 
     def setUp(self):
         self.spec = Spec()
+        with dnf.Base() as self._dnf_base:
```
Comment (jsilhan) [inline tests/test_plugins.py]: actually I'd prefer to mock dnf.Query.filter to return fake Package object in `test_files_to_pkgs` so we don't need to wait for DNF to download metadata.

File: tests/test_plugins.py (hunk 1 above)
Comment (jsilhan) [inline tests/test_plugins.py]: The real testing with DNF should be executed in `FunctionalTest.test_c_project`. Can you add the tests here too, please.

**Comment by cibo94** [comment]
You're right in https://github.com/rh-lab-q/rpg/pull/164#issuecomment-114091345, but dnf is always finding newest packages. So if we add ```>= version``` to the required package, older versions of OS (than the one we are building package on) may throw error even if they have right (newest) version.

Other matters  will be solved soon.

**Comment by cibo94** [comment]
And I am picking unique packages because some files may be from the same package.

**Comment by cibo94** [comment]
If I will test ```len(self.spec.Requires) == 1```, it will fail because there are actually 2 required packages:
```python
['python3-dnf >= 1.0.0', 'python3-dnf >= 1.0.1']
```
So should files_to_pckgs pick the higher version?

**Comment by jsilhan** [comment]
well, pick just name - it will be more simple to read and debug. We can improve it later.

**Comment by jsilhan** [comment]
Everything exept b471e03 merged, thanks

File: tests/test_plugins.py [hunk 3]
```diff
@@ -48,7 +69,16 @@ def test_find_files(self):
                  ('/archives/sample.tar.xz', None, None),
                  ('/Makefile', None, None),
                  ('/py/requires/sourcecode2.py', None, None)]
-        sorted_files = sorted(files, key=lambda e: e[0])
+        excludes = [('/patch/__pycache__/', r'%exclude', None),
```
Comment (jsilhan) [inline tests/test_plugins.py]: No. We want make `.../__pycache__` dir only for `....py` file not for every file.

**Comment by jsilhan** [comment]
Merged all but f33b92a. Thanks

**Comment by jsilhan** [comment]
Merged, thanks

**Pull request closed without merging**